cogboard-app/src/main/kotlin/com/cognifide/cogboard/widget/type/JiraBucketWidget.kt

```diff
@@ -0,0 +1,40 @@
+package com.cognifide.cogboard.widget.type
+
+import com.cognifide.cogboard.widget.AsyncWidget
+import io.vertx.core.Vertx
+import io.vertx.core.json.JsonArray
+import io.vertx.core.json.JsonObject
+import com.cognifide.cogboard.CogboardConstants as CC
+
+class JiraBucketWidget(vertx: Vertx, config: JsonObject) : AsyncWidget(vertx, config) {
+
+    private val bucketQueries: JsonArray = config.getJsonArray("bucketQueries")
+    private val bucketNames: JsonArray = JsonArray()
+    private val issueCounts: JsonArray = JsonArray()
+
+    override fun handleResponse(responseBody: JsonObject) {
+        val response = JsonObject()
+        val issues = responseBody.getJsonArray("issues").size()
+
+        bucketNames.add(responseBody.getString("requestId"))
+        issueCounts.add(issues)
+        response.put("bucketNames", bucketNames)
+        response.put("issueCounts", issueCounts)
+        send(JsonObject()
+                .put(CC.PROP_CONTENT, response))
```

Comment:
Instead of sending two objects that depend on each other order ... model response like this:
response example:
```
{
  "requestId_1": {
    "name": "Bucket 1 Name",
    "itemsCount": 5
  },
"requestId_2": {
    "name": "Bucket 2 Name",
    "itemsCount": 10
  },
}
```

Reply:
changed